Create national-natural-science-foundation-of-china.csl

jiongdd opened the pull request.

rmzelle commented:
@jiongdd, is this a different format than that of the existing https://www.zotero.org/styles?q=gb7714 styles? The "documentation" link in your style points to that style guide.

jiongdd commented:
@rmzelle Thanks to answer me. I made some changes on their basis to meet the requirements of NSFC(National Natural Science Foundation of China). I didn't notice the "documentation" link when I write this.

rmzelle commented:
@jiongdd, is there an online description of this style? We usually require a URL that points to a description of the style format.

And who would be using this style? Is it e.g. for writing NSFC grant proposals, or something else?

jiongdd commented:
@rmzelle yes, it's for writing the project requisition of NSFC. 

But there is not an online description of this style. The NSFC only requires that the reference format provides all authors. But I found an example of citation style in the document of contributors templates. You can download it using the following URL. But it's in Chinese. https://isisn.nsfc.gov.cn/egrantres/template/person/%E5%8F%82%E4%B8%8E%E4%BA%BA%E6%A8%A1%E6%9D%BF-%E9%80%9A%E7%94%A8.doc

I made this style on the basis of the example. What should I do to the documentation URL in my situation.

rmzelle commented:
Thanks. Is there an NSFC webpage that links to that Word document? (we prefer to not link to Word documents directly)

jiongdd commented:
@rmzelle I found the Word document under a tab in the Internet-Based Science Information System when I login. So I'm afraid that I can just offer you the link to the ISIS system because there isn't an NSFC webpage that links to that Word document. 
https://isisn.nsfc.gov.cn/egrantweb/main

rmzelle commented:
@jiongdd, sorry for letting this sit. Thanks for the explanation. Linking to the Word document is fine, in that case.

Since the CSL style is in Chinese (because of `default-locale="zh-CN"`), would it make sense to change the style title from English to Chinese? Would that make the style easier to find?

We also usually don't include the year in style file names and titles.

rmzelle commented on national-natural-science-foundation-of-china-2017.csl:
```diff
@@ -0,0 +1,206 @@
+  </info>
+  <macro name="author">
+    <names variable="author">
+      <name delimiter-precedes-last="always" et-al-min="999" et-al-use-first="999" initialize-with=" " name-as-sort-order="all" sort-separator=" ">
```
@jiongdd, "et-al-use-first" (the number of authors to show before "et al." [and others]) must always have a lower value than "et-al-min" (the number of authors beyond which name lists are abbreviated).

jiongdd replied:
I set the "et-al-use-first" and "et-al-min" a big number to meet the NSFC's require that the reference format provides all authors. I changed the "et-al-use-first" to 800, it's that OK?

rmzelle commented on national-natural-science-foundation-of-china-2017.csl:
```diff
@@ -0,0 +1,206 @@
+    <category field="engineering"/>
+    <category field="science"/>
+    <category field="generic-base"/>
+    <summary>This style just partly implemented what the National Natural Science Foundation of China requires.</summary>
```
> This style just partly implemented what the National Natural Science Foundation of China requires.

@jiongdd, can you explain a little what you mean by this? Did you have trouble getting the style right, or did you only check it for a low number of different item types (like articles, books, book chapters, etc.)?

jiongdd replied:
I removed the word "partly". I want to explain what this style is writing for.
I think this style have meet all the NSFC's requirement.

jiongdd commented:
@rmzelle I found that there isn't any style whose name is in Chinese in the Zotero Style Repository. So I though the style name must be written in English. I have removed the year in the style file name and title.
I set the "et-al-use-first" and "et-al-min" a big number to meet the NSFC's require that the reference format provides all authors.  I changed the "et-al-use-first" to 800, it's that OK?

rmzelle commented:
> I found that there isn't any style whose name is in Chinese in the Zotero Style Repository. So I though the style name must be written in English.

For styles that are used in another language, both that language and English are acceptable for the style title. We don't have a lot of Chinese styles, but see e.g. https://www.zotero.org/styles?q=(russian) and https://www.zotero.org/styles?q=id%3Athe-institute-of-electronics-information-and-communication-engineers.

rmzelle commented:
Great, thanks!

jiongdd commented:
@rmzelle Thanks very much!
